File: t2.py

import time
import numpy as np

# Генератор
def generator(n):
    for i in range(n):
        yield i

start_time = time.time()
sum_gen = sum(generator(int(1e7)))
print(f"Generator: {time.time() - start_time} seconds")

# Множення матриць з NumPy
start_time = time.time()
arr1 = np.random.rand(1000, 1000)
arr2 = np.random.rand(1000, 1000)
result = np.dot(arr1, arr2)
print(f"NumPy matrix multiplication: {time.time() - start_time} seconds")

# Сортування з NumPy
start_time = time.time()
arr = np.random.rand(int(1e7))
sorted_arr = np.sort(arr)
print(f"NumPy sort: {time.time() - start_time} seconds")

# Читання великого файлу
start_time = time.time()
with open('large_file.txt', 'r') as f:
    data = f.read()
print(f"File read: {time.time() - start_time} seconds")

File: t2.js

const fs = require('fs');

// Генератор
function* generator(n) {
    for (let i = 0; i < n; i++) {
        yield i;
    }
}

console.time('Generator');
let sumGen = 0;
for (let value of generator(1e7)) {
    sumGen += value;
}
console.timeEnd('Generator');

// Множення матриць
console.time('Matrix multiplication');
let arr1 = Array.from({ length: 1000 }, () => Array.from({ length: 1000 }, () => Math.random()));
let arr2 = Array.from({ length: 1000 }, () => Array.from({ length: 1000 }, () => Math.random()));
let result = Array.from({ length: 1000 }, () => Array(1000).fill(0));

for (let i = 0; i < 1000; i++) {
    for (let j = 0; j < 1000; j++) {
        for (let k = 0; k < 1000; k++) {
            result[i][j] += arr1[i][k] * arr2[k][j];
        }
    }
}
console.timeEnd('Matrix multiplication');

// Сортування
console.time('Array sort');
const arr = Array.from({ length: 1e7 }, () => Math.random());
arr.sort((a, b) => a - b);
console.timeEnd('Array sort');

// Читання великого файлу
console.time('File read');
const data = fs.readFileSync('large_file.txt', 'utf8');
console.timeEnd('File read');
